cmd/nerdctl/container_run.go

```diff
@@ -724,12 +733,36 @@ func createContainer(ctx context.Context, cmd *cobra.Command, client *containerd
 
 	var s specs.Spec
 	spec := containerd.WithSpec(&s, opts...)
+
 	cOpts = append(cOpts, spec)
 
-	container, err := client.NewContainer(ctx, id, cOpts...)
-	if err != nil {
+	container, cerr := client.NewContainer(ctx, id, cOpts...)
+	var netErr error
+	// NOTE: on non-Windows platforms, network setup is performed by OCI hooks.
+	// Seeing as though Windows does not currently support OCI hooks, we must explicitly
+	// perform the network cleanup in the main nerdctl executable.
+	if runtime.GOOS == "windows" {
+		netErr = netManager.SetupNetworking(ctx, id)
+	}
+
+	if cerr != nil || netErr != nil {
+		err := cerr
+		if err == nil {
+			err = netErr
+		}
```

Comment:
I am not sure I have an answer but all of the error handling here feels complex with lots of branching which could be hard to maintain.

Reply:
Agreed, will try to make the logic more legible.